src/lib/comms/sol-http-client-impl-curl.c

```diff
@@ -180,11 +180,11 @@ call_connection_finish_cb(struct sol_http_client_connection *connection)
     response->param = connection->response_params;
     response->response_code = (int)response_code;
 
-    connection->cb((void *)connection->data, connection, response);
+    connection->interface->response_cb((void *)connection->data, connection, response);
```

Comment:
it looks like you don't check if given `interface->response_cb` (or the `cb` in the version that takes a single cb instead of an interface) are NULL, then it would crash here. Either check it here or at the `sol_http_client_request()`  and `sol_http_client_request_with_interface()`